src/orion/core/io/experiment_builder.py

```diff
@@ -630,21 +630,22 @@ def _instantiate_algo(
 
     """
     config = config or orion.core.config.experiment.algorithms
-    if isinstance(config, str):
-        algo_type = algo_factory.get_class(config)
-        algo_config = {}
-    elif inspect.isclass(config) and issubclass(config, BaseAlgorithm):
-        algo_type: type[BaseAlgorithm] = config
-        algo_config = {}
-    elif isinstance(config, dict):
-        backported_config = backward.port_algo_config(config)
-        algo_name = backported_config.pop("of_type")
-        algo_type = algo_factory.get_class(algo_name)
-        algo_config = {}
-    else:
-        raise ValueError(f"Invalid algorithm configuration: {config}")
 
     try:
+        if isinstance(config, str):
+            algo_type = algo_factory.get_class(config)
+            algo_config = {}
```

Comment:
Looks like the if-else clauses here are not tested.

Reply:
You mean you'd rather have those as a sequence of ifs, rather than elifs?
If so, even though you're correct that it would work with a sequence of ifs, since they don't affect each other's conditions, I like the clarity that elifs bring: It's 100% unambiguous that only one condition will be met.

Reply:
Oh no! Elifs are better! I meant that the if-else clauses (including the elifs) are currently not covered by the unit-tests.

Reply:
> Looks like the if-else clauses here are not tested.

I'm sorry, I misread that as "looks like the if-else clauses are not used"!

Reply:
Haha, no problem!

Reply:
Ok there's something weird with the coverage I think. Last time I checked, it was saying that none of the cases were being tested.

Reply:
Yeah it should go through at least one of them. :thinking: I am pretty sure `str` and `dict` is covered. Not sure about BaseAlgorithm and the else-clause (no algo found).

Reply:
Should be fixed in 5040c07f67c31da2d7bac6a74cfec256d2ae574a